### ISSUE TITLE
`ToDocument`: `char` values are printed wrongly

### ISSUE THREAD

**Issue by W95Psp** [FStarLang/FStar#3560]
Chars are printed incorrectly by F*'s pretty printer. Consider:
```fstar
module A
open FStar.Char

let _ = ['\0'; '\n'; 'é'; 'ê'; 'ñ']
```

`fstar.exe --print A.fst`:
```fstar
module A
open FStar.Char

let x = ['\000'; 0x0Az; '\233'; '\234'; '\241']
```
`\000`,  `\233`, `\234`, `\241` are all not accepted by F*'s lexer

And `0x0Az` is a uint 8 literal.

This is mostly because we use directly `PPrint.OCaml.char`, which assumes that's OCaml that will parse back the result. We should implement our own `char`, that could use unicode.

Note: those bugs are present on string literals as well.

**Comment by mtzguido** [comment]
Unsure why this didn't autoclose but:

Fixed by #3561, thanks Lucas!

**Comment by mtzguido** [comment]
Unsure why this didn't autoclose but:

Fixed by #3561, thanks Lucas!
